ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -1778,6 +1778,20 @@ static IHqlExpression * simplifySortlistComplexity(IHqlExpression * sortlist)
                     appendComponent(cpts, invert, &concats.item(idxc));
             }
         }
+        else if (cur->getOperator() == no_trim)
+        {
+            //TRIM(fixed-length-string) can just sort by the string instead.  (Don't match LEFT/RIGHT versions.)
```

Comment:
If two strings are unequal length the longer one is compared as if the shorter one is implicitly padded with spaces.

I think it shouldn't be restricted to fixed length strings...  however the issue is that if they are variable length strings, one is 'AB' and the other is 'AB ' this would optimize away a trim, which would then possibly make the sort be marked as unstable - even though theoretically the fields could be distinguished because of the length.

On reflection, I this should allow variable length strings.  Instead the code that checks if the dataset is sorted by all fields should theoretically not match any fields that are variable length strings because the comparison operator doesn't compare all the information from the field.

We may prefer to ignore that issue, or at least have an option that defaults to ignoring it.